### ISSUE TITLE
Donut chart default color pattern has only 2 colors

### ISSUE THREAD

**Issue by amirfefer** [patternfly/patternfly-3#902]
`blue` and `black-300` via [getDefaultDonutColors](https://github.com/patternfly/patternfly/blob/f3e97eed896f1cae0402a86ee526b68fc7abbb96/src/js/patternfly-settings-charts.js#L150).
Shouldn't we add a 6 colors (or more) pattern?

**Comment by dabeng** [comment]
Hi @amirfefer , I don't think this is a issue.

According to the [design doc ](http://www.patternfly.org/pattern-library/data-visualization/donut-chart/#/design)of utilization donut chart, it only include two factors(used & available), so two colors ` patternfly.pfPaletteColors.blue` `patternfly.pfPaletteColors.black300` are just the right amount.

Of course, you can specify the colors for your utilization donut chart with following code snippets:
```js
  donutConfig.color = { pattern: [
      $.pfPaletteColors.red,
      $.pfPaletteColors.green,
    ]
  };
```

**Comment by ohadlevy** [comment]
I would think it would be nice if we had a default colors (maybe the same
as in the design docs). As it stands now it's inconsistent as for one type
you have a default vs the other which doesn't.

**Comment by jeff-phillips-18** [comment]
Agreed @ohadlevy. Based on https://github.com/theforeman/foreman/pull/5137#issuecomment-356959608, I think the default colors ought to be:

            patternfly.pfPaletteColors.blue,
            patternfly.pfPaletteColors.black300
            patternfly.pfPaletteColors.red100
            patternfly.pfPaletteColors.orange400
            patternfly.pfPaletteColors.green400
            patternfly.pfPaletteColors.cyan500
            patternfly.pfPaletteColors.gold200

The first 2 need to be what is already defined as to not break anyone currently using 2 default colors.

**Comment by dabeng** [comment]
@jeff-phillips-18 , you mean we should remove the following settings
```js
    colors: {
      Cats: $.pfPaletteColors.blue,
      Hamsters: $.pfPaletteColors.green,
      Fish: $.pfPaletteColors.orange,
      Dogs: $.pfPaletteColors.red
    },
````
> https://github.com/patternfly/patternfly/blob/master/tests/pages/_includes/widgets/charts/donut-whole-relationship.html#L9~L14

> https://github.com/patternfly/patternfly/blob/master/tests/pages/_includes/widgets/charts/donut-mini.html#L8~L13

and replace the following settings
```js
    getDefaultDonutColors = function () {
        return {
          pattern: [
            patternfly.pfPaletteColors.blue,
            patternfly.pfPaletteColors.black300
          ]
        };
      },
```
with
```js
    getDefaultDonutColors = function () {
        return {
          pattern: [
            patternfly.pfPaletteColors.blue,
            patternfly.pfPaletteColors.black300,
            patternfly.pfPaletteColors.red100,
            patternfly.pfPaletteColors.orange400,
            patternfly.pfPaletteColors.green400,
            patternfly.pfPaletteColors.cyan500,
            patternfly.pfPaletteColors.gold200,
          ]
        };
      },
```

**Comment by sharvit** [comment]
@dabeng @jeff-phillips-18 @amirfefer 
Based on this document: http://www.patternfly.org/pattern-library/data-visualization/donut-chart/

It feels like you have 2 different set of donut-charts:
* UtilizationDonutChart - 2 colors - represent a scale of 1 value from 0% to 100%
* RelationshipDonutChart - up to 6 colors - represent relationship between values

I don't think that the `black300` should be used in the `RelationshipDonutChart` because it represent no-data/empty space.

I understand it should be something like:
```js
const getDefaultDonutColors = (utilization = false) = {
  if (utilization) return getDefaultUtilizationDonutColors();
  return getDefaultRelationshipDonutColors();
};

const getDefaultRelationshipDonutColors = () => ({
  pattern: [
            patternfly.pfPaletteColors.blue,
            patternfly.pfPaletteColors.red100,
            patternfly.pfPaletteColors.orange400,
            patternfly.pfPaletteColors.green400,
            patternfly.pfPaletteColors.cyan500,
            patternfly.pfPaletteColors.gold200,
  ]
});
const getDefaultUtilizationDonutColors = () => ({
  pattern: [
            patternfly.pfPaletteColors.blue,
            patternfly.pfPaletteColors.black300
  ]
});
```

**Comment by sharvit** [comment]
@jeff-phillips-18 @yaacov
Do you think `patternfly-react` should export 2 different components for DonutChart? 

So i could do so without worry about the default values:
```js
import { UtilizationDonutChart, RelationshipDonutChart ) from 'patternfly-react';
```

**Comment by jeff-phillips-18** [comment]
@sharvit Good point about the black300 not being used for non-utilization donuts.
I like the idea of having the capability for both as you suggest above.

As for `patternfly-react`, it could use multiple components or the existing DonutChart component could take a property to specify the chart type.

**Comment by sharvit** [comment]
@jeff-phillips-18 I have talked with @amirfefer and we both agreed about:
* `DonutChart` component should default act as a `RelationshipDonutChart`
* Create a `UtilizationDonutChart` component that render `DonutChart` with utilization colors.

**Comment by yaacov** [comment]
> @jeff-phillips-18 I have talked with @amirfefer and we both agreed about:
>
>    DonutChart component should default act as a RelationshipDonutChart
>    Create a UtilizationDonutChart component that render DonutChart with utilization colors.

I like this approach too 👍

**Comment by jeff-phillips-18** [comment]
Works for me, though out of context for this particular repo/issue 😉

**Comment by dabeng** [comment]
Hi @jeff-phillips-18 @sharvit , I have committed PR #916 for this issue. I think I adopted the least-cost solution that I just provide override color option for "available" data column of utilization donut chart and add more default colors for  all donut charts.

**Comment by sharvit** [comment]
Thanks @dabeng :+1:

**Comment by amirfefer** [comment]
Thanks @dabeng ! :+1:

**Comment by jeff-phillips-18** [comment]
Fixed by #916
